File: Legacy/Shared/Resources/bogosort.bundle/bogosort.py
import random

def sort(arr):
  expected = arr[:]
  expected.sort()
  while not (arr == expected):
    random.shuffle(arr)

if __name__ == "__main__":
  array = [0, 39, 21, 62, 91, 77, 14, 23]
  sort(array)
  print(array)

File: App/Resources/Algorithms/bogosort.js
// Faithfully mirrors ArrayV's bogoSwap: a full Fisher-Yates reshuffle each iteration,
// not a single random swap.
function sort(engine) {
    const n = engine.count();

    function isSorted() {
        for (let i = 1; i < n; i++) {
            if (!engine.compare(i, i - 1)) return false;
        }
        return true;
    }

    function shuffle() {
        for (let i = 0; i < n; i++) {
            const j = i + Math.floor(Math.random() * (n - i));
            engine.swap(i, j);
        }
    }

    while (!isSorted()) {
        shuffle();
    }
}
